TeaLeaf/src/com/tealeaf/TeaLeaf.java

```diff
@@ -303,18 +304,25 @@ public void onCreate(Bundle savedInstanceState) {
 
 		// default screen dimensions
 		Display display = getWindow().getWindowManager().getDefaultDisplay();
+		Point screenSize = new Point();
 		int width = display.getWidth();
 		int height = display.getHeight();
 		int orientation = getRequestedOrientation();
 
 		// gets real screen dimensions without nav bars on recent API versions
-		if (isFullScreen && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-			Point screenSize = new Point();
+		if (isFullScreen && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT && !notched_device) {
```

Comment:
ideally !notched_device variable is not needed
isNotchedDevice() should suffice. 

But let's go with this for now.